### ISSUE TITLE
SnagIt Capture fails in SnagIt 2024

### ISSUE THREAD

**Issue by RickStrahl** [RickStrahl/MarkdownMonster#1157]
SnagIt Integrated Capture fails when trying to Finish the capture in the SnagIt editor.

![image](https://github.com/user-attachments/assets/ca5404ce-c541-4379-8f75-5564c67e7a62)

**Comment by RickStrahl** [comment]
I was able to duplicate this issue referenced in #1155 with SnagIt 2024. 

Unfortunately this appears to be an issue with SnagIt and not with Markdown Monster's integration since the capture fails from within SnagIt.

MM uses the Clipboard capture option (mode 4) to capture the output from the capture window into the clipboard, and then offers on the MM side to capture and paste the image. However, the SnagIt COM Capture never actually returns as it fails in the actual COM call inside of SnagIt.

In the past these issues were caused by having mutliple instances of SnagIt installed or uininstalling one and ending up with incorrect COM references. However, I tried uninstalling my older version SnagIt 2020 and then re-registering the 2024 version and the problem persisted.

I've fought these issues for years with Techsmith in the past, but they had no interest in discussing these issues. Up to v2020 the COM integration was working correctly but it appears it is now broken again to the point where it is not directly usable.

I'm considering just removing support as this is becoming untenable as I can't control the third party control.

**Comment by RickStrahl** [comment]
## WORKAROUND FOR SNAGIT CAPTURE

There are a couple of workarounds for using the SnagIt Capture:

* Manually use SnagIt and Copy to Clipboard and Paste
* Use SnagIt Capture Integration but manually copy to Clipboard

### Manually Capture and Paste from SnagIt
This is the approach I personally use. I use SnagIt extensively and because I use it from many applications I tend to use the same approach for all of them which is as follows:

* Open SnagIt Capture (from task tray shortcut or Windows Key-SnagIt)
* SnagIt configured to open capture in the editor
* SnagIt configured to automatically copy to clipboard
* Start Capture
* Capture content
* Edit in editor if desired
* If changes were made Ctrl-A Ctrl-C to copy image to clipboard
* Switch back to MM
* Paste from Clipboard

To me this workflow works just as fast as the integrated SnagIt Image Capture .


### Use the SnagIt Image Capture, but don't Finish!
The SnagIt Image Capture actually works, but it's the `Finish` operation that blows up SnagIt. So, instead of clicking finish you can do the following in the SnagIt Editor:

* SnagIt Editor Opens with captured image
* Edit Image as needed
* Press Ctrl-A Ctrl-C to copy image to clipboard
* **CANCEL** out of the Image Editor ( **not Finish**)

At this point the SnagIt Image Capture window behaves the same as if you had selected finish. Essentially you are replacing the Finish button with `Ctrl-a`, `Cntrl-c`, Cancel and otherwise it behaves the same.

This obviously sub-optimal but until (if ever) SnagIt fixes the COM object, if you really want to use the integrated capture this is an easy workaround for the same workflow.

**Comment by SyntaxDreamer** [comment]
Confirmed, manually copying works fine and then pasting the image. It does seem an issue with Snagit when clicking the Finish button.

**Comment by RickStrahl** [comment]
So I gave this another try today, after I had uninstalled **all** of my versions of SnagIt and rebooted, then reinstalled the version I'd been using (SnagIt 2021).

At this point SnagIt is working again for me in the captures.


If I remember this correctly from days back, the issue is that if multiple versions of SnagIt have been installed there are several registry references to the COM objects and apparently Techsmith is not updating the registration data correctly.

Uninstalling and reinstalling every version, rebooting then reinstalling Snagit is what worked for me.

As I mentioned I had bugged this years ago on their tech support forum and Techsmith acknowledged but never bothered to follow up or apparently fix this. This is probably very low priority for them, but it sucks.

At the end of the day it's just as easy to use SnagIt to Clipboard directly as opposed to this addin, but now that it works I do use it since it saves a couple of mouse clicks and the flipping back and forth.
